### ISSUE TITLE
Can you provide an offical demo? I have write one.

### ISSUE THREAD

**Issue by Originlightwkp** [SunzeY/AlphaCLIP#7]
`import alpha_clip
from alpha_clip.simple_tokenizer import SimpleTokenizer
import torch
from torchvision import transforms

from PIL import Image
import clip
import numpy as np

device = "cuda" if torch.cuda.is_available() else "cpu"
alpha_clip_modelname = "ViT-L/14"
alpha_clip_model, alpha_clip_preprocess = alpha_clip.load("ViT-L/14", alpha_vision_ckpt_pth="checkpoints/clip_l14_grit+mim_fultune_6xe.pth")
alpha_clip_model = alpha_clip_model.float().to(device)

alpha_clip_mask_transform = transforms.Compose([
    transforms.ToTensor(),
    transforms.Resize((224, 224)),
    transforms.Normalize(0.5, 0.26)
])

img_path = 'assets/maxi_dress.jpg'
mask_path = 'assets/1.png'

raw_img = Image.open(img_path)
mask = Image.open(img_path)
mask = np.array(mask)
mask = torch.tensor(mask)
mask = mask[:,:,0]
mask = mask.unsqueeze(2)
#print(mask.shape)
mask = np.array(mask)

#print(mask.shape)
alpha_map = alpha_clip_mask_transform(mask * 255).to(device,dtype=torch.float).unsqueeze(0)
#print(alpha_map.shape)
#alpha_map = alpha_map[0,:,:]
#alpha_map = alpha_map.unsqueeze(0)
#print(alpha_map.shape)
image = alpha_clip_preprocess(raw_img).unsqueeze(0).to(device, dtype=torch.float)
#print(image.shape)
image_features = alpha_clip_model.visual(image, alpha_map) # This works, I have image_features vector of size [1,768]
print(image_features.shape)`

**Comment by Originlightwkp** [comment]
import alpha_clip
from alpha_clip.simple_tokenizer import SimpleTokenizer
import torch
from torchvision import transforms
torch.set_printoptions(precision=4,sci_mode=False)
from PIL import Image
import clip
import numpy as np

device = "cuda" if torch.cuda.is_available() else "cpu"
alpha_clip_modelname = "ViT-L/14"
alpha_clip_model, alpha_clip_preprocess = alpha_clip.load("ViT-L/14", alpha_vision_ckpt_pth="checkpoints/clip_l14_grit+mim_fultune_6xe.pth")
alpha_clip_model = alpha_clip_model.float().to(device)

alpha_clip_mask_transform = transforms.Compose([
    transforms.ToTensor(),
    transforms.Resize((224, 224)),
    transforms.Normalize(0.5, 0.26)
])

img_path = 'assets/maxi_dress.jpg'
mask_path = 'assets/1.png'

raw_img = Image.open(img_path)
mask = Image.open(img_path)
mask = np.array(mask)
mask = torch.tensor(mask)
mask = mask[:,:,0]
mask = mask.unsqueeze(2)
#print(mask.shape)
mask = np.array(mask)

#print(mask.shape)
alpha_map = alpha_clip_mask_transform(mask * 255).to(device,dtype=torch.float).unsqueeze(0)
#print(alpha_map.shape)
#alpha_map = alpha_map[0,:,:]
#alpha_map = alpha_map.unsqueeze(0)
#print(alpha_map.shape)
image = alpha_clip_preprocess(raw_img).unsqueeze(0).to(device, dtype=torch.float)
#print(image.shape)
image_features = alpha_clip_model.visual(image, alpha_map) # This works, I have image_features vector of size [1,768]
print(image_features.shape)
print(image_features.type())
model, preprocess = clip.load('ViT-L/14', device)
# I tried the follow to extract the text features:
#text_tokenized = tokenizer.encode('my great text')
#text_tokenized = torch.Tensor(text_tokenized ).type(torch.int).to(device)
text = clip.tokenize(["purple long sleeve dress", "purple sleeveless dress", "a flower"]).to(device)
text_features = model.encode_text(text).to(dtype=torch.float)
print(text_features.shape)
print(text_features.type())
with torch.no_grad():
    #image_features = model.encode_image(img)  # Input tensor should have shape (b,c,h,w)
    #text_features = model.encode_text(tokenized_sleevelength_prompt)
    image_features /= image_features.norm(dim=-1, keepdim=True)
    text_features /= text_features.norm(dim=-1, keepdim=True)

    text_probs = (100.0 * image_features @ text_features.T).softmax(dim=-1)
    print(text_probs)
    values, indices = torch.topk(text_probs, 1)

**Comment by SunzeY** [comment]
I'm not sure what's your meaning of demo, We have list three possible demo cases in readme:
Usage examples are available
* Visualization of attention map: [notebook](https://github.com/SunzeY/AlphaCLIP/blob/main/notebooks/attn_visual.ipynb)
* Alpha-CLIP used in BLIP-Diffusion: [notebook](https://github.com/SunzeY/AlphaCLIP/blob/main/notebooks/blip_diffusion.ipynb)
* Alpha-CLIP used in SD_ImageVar: [demo](https://github.com/SunzeY/AlphaCLIP/tree/main/demo/with_diffusion)

**Comment by Originlightwkp** [comment]
Thank you! I mean compute the sililarity between text and image.

**Comment by Aleafy** [comment]
Thank you for the code and examples you provided. We have released a demo for classification. Welcome more attempts and questions!
